### PR TITLE
Add User management table

### LINKED ISSUE #454: Users table header
### Summary and Background
Users table header tells the Super Admin what user information is displayed.

### User Story
As Super Admin, I want to see what user information is displayed.
As Super Admin, I want to be able to select all users in a table.

### Solution (Core Elements)
Create Users table header. Columns:
- Select box: On click, each row is selected. Clicking again undos these changes.
- First name: On click, rows are sorted in reverse direction (opposite alphabetical) and arrow points upward. Clicking again undos these changes.
- Last name
- Pronouns
- Email

### Mockups

Other designs: https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1643%3A48828

### Technical
- You will make your changes in `/admin/users` route
- You will need to use `react-table` to get the header click to change sorting order to work https://chakra-ui.com/getting-started/with-react-table this ticket will require you to setup the template for using `react-table` with chakra ui
- reference https://chakra-ui.com/getting-started/with-react-table for an example
- For this ticket, you should create a table with just a `Thread` component for all the necessary header columns

### PR DESCRIPTION
**Pull request by MatoPlus**
## Ticket link

Closes #454, #455, #540

## Implementation description
* Added user management table as designed via tanstack react-table
* Integrated search and pagination to table
* Integrated existing profile drawer to table (still need to hook up profile drawer to interact with API)
* Added new backend resolvers to update/append branches only, due to limited input in user management table
* All branch edit actions are fully implemented and integrated
* Full user management flow should be complete
* 
https://user-images.githubusercontent.com/44826218/182489354-86298682-6e56-4051-8409-3585fd8c5099.mp4

### Test queries for new resolvers

```gql
mutation {
  appendBranchesForMultipleUsersByEmail(emails: ["sistering+employee@uwblueprint.org", "sistering+employee1@uwblueprint.org"], branchIds: [1])
}
```

```gql
mutation {
  updateUserBranchesByEmail(email: "sistering+employee@uwblueprint.org", branchIds: [1])
}
```

## Steps to test
1. Login as a admin and go to `/admin/users` and interact with table. It should work as per designed

## What should reviewers focus on?
* Code cleanliness (will do further follow up in full integration ticket #540 )
* Code correctness

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/users/UserManagementTableProfileCell.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  emergencyContactName: string;
+  emergencyContactPhone: string;
+  emergencyContactEmail: string;
+  languages: string[];
+  skills?: string[];
+  isVolunteer?: boolean;
+  branches: BranchResponseDTO[];
+  selectedBranches: BranchResponseDTO[];
+  handleBranchMenuItemClicked: (item: BranchResponseDTO) => void;
+  // TODO: Replace param type
```
Comment (MatoPlus) [inline frontend/src/components/admin/users/UserManagementTableProfileCell.tsx:22]:
```suggestion
```

**Review by neelismail01 — APPROVED** [review]
Everything looks good to me. LGTM!

**Comment by MatoPlus** [comment]
Still trying to integrate new resolvers into frontend page

File: backend/services/implementations/branchService.ts [hunk 2]
```diff
@@ -119,6 +121,118 @@ class BranchService implements IBranchService {
...
+          });
+        }
+      });
+    } catch (error: unknown) {
+      Logger.error(`Failed to get user. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return true;
+  }
```
Comment (MatoPlus) [inline backend/services/implementations/branchService.ts:235]: This seems inefficient, open to suggestions to make it cleaner

**Comment by MatoPlus** [comment]
Now I should also follow up to show pronouns and date of birth for admins

**Comment by MatoPlus** [comment]
Gonna merge this for now as it has been approved before, don't wanna make this stale. For the reviewers, please let me know if you have any suggestions that comes up even after this gets merged.

**Pull request merged**